tools/mitre/mitredb.py

```diff
@@ -33,15 +35,152 @@ class Metadata(Base):
+    """
+    __tablename__ = "software"
+
+    Id = Column(const.ID_t, String, primary_key=True)
+    name = Column(const.NAME_t, String, nullable=False)
+    description = Column(const.DESCRIPTION_t, String, default=None)
+    created_time = Column(const.CREATED_t, DateTime, default=None)
+    modified_time = Column(const.MODIFIED_t, DateTime, default=None)
+    mitre_version = Column(const.MITRE_VERSION_t, String, default=None)
+    revoked_by = Column(const.REVOKED_BY_t, String, default=None)
+    deprecated = Column(const.DEPRECATED_t, Boolean, default=False)
+
+    def __init__(self, Id="", name="", description=None, created_time=None, modified_time=None, mitre_version=None, revoked_by=None, deprecated=False) :
+        self.Id = Id
+        self.name = name
+        self.description = description
+        self.created_time = created_time
+        self.modified_time = modified_time
+        self.mitre_version = mitre_version
+        self.revoked_by = revoked_by
+        self.deprecated = deprecated
+
+class Mitigations(Base):
+    """
+    In this table are stored the mitigations of json file
+    The information stored:
+        id: Used to identify the mitigation (PK)
+        name: Name of the mitigation
+        description: Detailed description of the mitigation
+        created_time: Publish date
+        modified_time: Last modification date
+        mitre_version: Version of MITRE when created
+        revoked_by: ID of the mitigation that revokes this one, NULL otherwise
+        deprecated: Boolean indicating if this mitigation is deprecated
+    """
+    __tablename__ = "mitigations"
+
+    Id = Column(const.ID_t, String, primary_key=True)
+    name = Column(const.NAME_t, String, nullable=False)
+    description = Column(const.DESCRIPTION_t, String, default=None)
+    created_time = Column(const.CREATED_t, DateTime, default=None)
+    modified_time = Column(const.MODIFIED_t, DateTime, default=None)
+    mitre_version = Column(const.MITRE_VERSION_t, String, default=None)
+    revoked_by = Column(const.REVOKED_BY_t, String, default=None)
+    deprecated = Column(const.DEPRECATED_t, Boolean, default=False)
+
+    def __init__(self, Id="", name="", description=None, created_time=None, modified_time=None, mitre_version=None, revoked_by=None, deprecated=False) :
+        self.Id = Id
+        self.name = name
+        self.description = description
+        self.created_time = created_time
+        self.modified_time = modified_time
+        self.mitre_version = mitre_version
+        self.revoked_by = revoked_by
+        self.deprecated = deprecated
+
+def parse_table_(function, data_object):
+    table = function()
+    table.Id = data_object[const.ID_j]
+    table.name = data_object[const.NAME_j]
+    if const.DESCRIPTION_j in data_object:
+        table.description = data_object[const.DESCRIPTION_j]
+
+    table.created_time = datetime.strptime(data_object[const.CREATED_j], '%Y-%m-%dT%H:%M:%S.%fZ')
```

Comment:
`created_time` and `modified_time` are always present? If so, make these parameters not nullable. If not, check if they are present in `data_object` before the assignment.